docs/user-guide/prototype-an-assistant.rst

```diff
@@ -0,0 +1,198 @@
+Prototype an Assistant
+======================
+
+.. edit-link::
+
+Get started with Rasa Open source and learn how to create an assistant from scratch!
+
+This page explains the basics of building an assistant with Rasa and shows the structure of a Rasa project.
+You can test it out right here without installing anything. You can also install Rasa and follow along in your command line.
+
+
+1. Define a basic user goal
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+To prototype an assistant, start with a single `user goal <glossary>`_
+that your assistant will handle. You can create a prototype using the user goal we have chosen
+"subscribing to a newsletter", or you can come up with your own. If you choose your own user goal,
+you can create a prototype that handles your goal by following the instructions to change each
+section throughout the tutorial.
+
+
+2. Create some NLU data
+~~~~~~~~~~~~~~~~~~~~~~~
+
+`NLU <glossary>`_ data provides examples of what users might say to your assistant and what they mean by it.
+`Intent <glossary>`_  refers to what the user means with a specific message.
+Your assistant can only learn to recognize intents for which it has seen examples, so we need to provide some data.
+
+Add examples for your user-goal specific intents in the format shown below.
+You can delete the ``subscribe`` and ``inform`` intents if you’re not using them;
+you can also add or change examples for any of the other intents.
+
+.. code-editor::
+    :language: yaml
+    :id: nlu
+    :height: 400
+    :tracking_endpoint: https://trainer-service.prototyping.rasa.com/startPrototyping
+
+    nlu:
+    - intent: greet
+      examples: |
+           Hi
+           Hey!
+           Hallo
+           Good day
+           Good morning
+
+    - intent: subscribe
+      examples: |
+           I want to get the newsletter
+           Can you send me the newsletter?
+           Can you sign me up for the newsletter?
+
+    - intent: inform
+      examples: |
+           My email is example@example.com
+           random@example.com
+           Please send it to anything@example.com
+           Email is something@example.com
+
+:ref:`Read about best practices for NLU <training-data-format>`
+
+
+3. Define a simple form
+~~~~~~~~~~~~~~~~~~~~~~~
+
+For most user goals, the bot will need to collect some information from the user to fulfill their request.
+To do so, we define a `form <glossary>`_. You can change the name of the form to reflect your user goal.
+Add to or replace the "email" item in the list below to reflect the information the bot needs
+to collect for your user goal. Leave the "type" field the same for any items you add.
+
+.. code-editor::
+    :language: yaml
+    :id: forms
+    :height: 150
+    :tracking_endpoint: https://trainer-service.prototyping.rasa.com/startPrototyping
+
+    forms:
+    - newsletter_form:
+       email:
+       - type: from_text
```

Comment:
```suggestion
    forms:
    - newsletter_form:
        email:
        - type: from_text
```
```suggestion
    forms:
    - newsletter_form:
        email:
        - type: from_text
```

Reply:
slash maybe in general we need to pick an indentation level and stick to it (there was 1, 2, and 4 i believe)